### PR TITLE
Update `config_machines.xml`

### PR DESCRIPTION
**Pull request by altheaden**
This PR is just to update the `config_machines.xml` file from [E3SM](https://github.com/E3SM-Project/E3SM/blob/master/cime_config/machines/config_machines.xml). I also updated some of the values we have in the files for each of the machines, as needed.

Checklist
* [ ] `Testing` comment in the PR documents testing used to verify the changes

### CODE REVIEW

**Comment by altheaden** [comment]
@xylar I set this PR to "in progress" for now because there are some machine files I haven't looked at yet to see if some of the older changes should be implemented. I'll put some notes here for you to look at and we can discuss.

**Comment by xylar** [comment]
Note that there are some more recent changes to include:
https://github.com/E3SM-Project/E3SM/pull/6596
https://github.com/E3SM-Project/E3SM/pull/6603

**Comment by xylar** [comment]
@altheaden, this looks excellent so far!

**Comment by altheaden** [comment]
@xylar I just rolled in the changes from the two E3SM PRs you mentioned and updated the appropriate files. I think all the spack files should be up to date now, along with `config_machines.xml`.

**Comment by xylar** [comment]
@altheaden, this all looks great to me! I think doing thorough testing of spack builds now using these changes will be too time consuming to be worthwhile.  So my suggestion is that we merge these changes and then test them thoroughly the next time we need to update E3SM-Unified, Polaris or Compass.

I'm testing things out on Compy right now so we'll at least be able to see if those changes work.

File: mache/spack/compy_intel_impi.yaml [hunk 1]
```diff
@@ -129,22 +129,22 @@ spack:
       buildable: false
     netcdf-c:
       externals:
-      - spec: netcdf-c~mpi~parallel-netcdf@4.6.3
-        prefix: /share/apps/netcdf/4.6.3/intel/19.0.5
+      - spec: netcdf-c+mpi~parallel-netcdf@4.6.3
+        prefix: /share/apps/netcdf/4.6.3/intel/20.0.0
```
Comment (xylar) [inline mache/spack/compy_intel_impi.yaml:133]: @altheaden, can you use a hard-reset to remove the last commit I made?  It turns out not to be correct.  NetCDF is still being built without MPI support.

Comment (altheaden) [reply mache/spack/compy_intel_impi.yaml:133]: Sure thing!

**Review by xylar — APPROVED** [review]
I think this looks good. As long as you think it's done, let's go ahead and merge it.

File: mache/spack/compy_intel_impi.yaml (hunk 1 above)
Comment (xylar) [reply mache/spack/compy_intel_impi.yaml]: Thanks!

**Comment by altheaden** [comment]
Sure, go ahead!

**Pull request merged**